### PR TITLE
Fixed some leafnode issues introduced from JS cluster work

### PR DESCRIPTION
**Pull request by kozlovic**
Also fixed a flapper.

Signed-off-by: Ivan Kozlovic <ivan@synadia.com>

### CODE REVIEW

File: server/leafnode.go [hunk 1]
```diff
@@ -1218,7 +1229,15 @@ func (c *client) processLeafNodeConnect(s *Server, arg []byte, lang string) erro
 	c.headers = supportHeaders && proto.Headers
 
 	// Remember the remote server.
-	c.leaf.remoteServer = proto.Name
+	// We changed the leafnode CONNECT.Name json tag from "name" to "server_name",
```
Comment (kozlovic) [inline server/leafnode.go:1232]: Note: I could remove this comment and code (just have c.leaf.remoteServer = proto.Name) if I revert the change from "name" to "server_name" that you did in JSC.

**Review by derekcollison — APPROVED** [review]
LGTM

File: server/leafnode.go (hunk 1 above)
Comment (derekcollison) [reply server/leafnode.go:1232]: ok sounds good.

Comment (kozlovic) [reply server/leafnode.go:1232]: Just double checking: sounds good mean that you agree on changing back "server_name" json tag to "name"?
